.changeset/long-apes-admire/changes.md

```diff
@@ -0,0 +1,38 @@
+TypeScript types have been restructured. These changes:
+
+- Reduce build times when using emotion
+- In many cases remove the need for manually specifying generic parameters for your emotion components.
+
+If you encounter build issues after upgrade, try removing any manually specified generic types and let them be inferred. Otherwise refer to the breaking changes list below.
+
+## Improvements
+
+- useTheme added to EmotionTheming interface and can now create your own closed variation of withTheme. More information in the docs under the theming section.
+- Union types as props are better supported and should be inferred properly
+- Build times should be reduced significantly in larger projects.
+
+## Breaking changes
+
+- Function interpolation can no longer return another function directly
```

Comment:
more out of curiosity than from wanting to allow it - does this change have any impact? as we've noticed that interpolation~ type still stays recursive, even when support for this one got dropped

Reply:
Not really, I could revert this change.

Reply:
Maybe let's revert it for now if it doesn't have any meaningful impact - less breaking changes and technically more in line with runtime semantics

Reply:
Done